### ISSUE TITLE
fb2: garbled images

### ISSUE THREAD

**Issue by mergen3107** [koreader/koreader#12212]
* KOReader version: 2024.07 and 2024.07-25
* Device: Kindle Scribe (5.16.2)

#### Issue
Images are garbled or missing (empty box icon) in a fb2 book.
They were fine some time before, around maybe 2024.04-150 to -200.

##### Screenshot:
![fb2_img_crash](https://github.com/user-attachments/assets/9db6bdea-848c-44b1-8aa2-d03e9dfe6f85)

##### `crash.log` (if applicable)
[fb2_img_crash.txt](https://github.com/user-attachments/files/16354676/fb2_img_crash.txt)

##### Book:
[Борис Акунин - Алмазная Колесница.fb2.zip](https://github.com/user-attachments/files/16354677/-.fb2.zip)

**Comment by benoit-pierre** [comment]
No images for me, but a bunch of related traces:
```
Corrupt JPEG data: 129 extraneous bytes before marker 0x79
Corrupt JPEG data: 1840 extraneous bytes before marker 0xd9
Corrupt JPEG data: 133 extraneous bytes before marker 0xc0
Corrupt JPEG data: 128 extraneous bytes before marker 0x6a
Corrupt JPEG data: 2334 extraneous bytes before marker 0xd9
Corrupt JPEG data: 146 extraneous bytes before marker 0x8d
Corrupt JPEG data: 1070 extraneous bytes before marker 0xd5
Corrupt JPEG data: 250 extraneous bytes before marker 0x8b
Corrupt JPEG data: 2395 extraneous bytes before marker 0xd9
Corrupt JPEG data: 204 extraneous bytes before marker 0x31
Corrupt JPEG data: 976 extraneous bytes before marker 0xd9
Corrupt JPEG data: 412 extraneous bytes before marker 0xc0
Corrupt JPEG data: 92 extraneous bytes before marker 0xc2
Corrupt JPEG data: 52 extraneous bytes before marker 0x85
Corrupt JPEG data: 388 extraneous bytes before marker 0xe3
Corrupt JPEG data: 87 extraneous bytes before marker 0x4a
Corrupt JPEG data: 445 extraneous bytes before marker 0xc9
Corrupt JPEG data: 5678 extraneous bytes before marker 0xd9
Corrupt JPEG data: 3275 extraneous bytes before marker 0xd9
Corrupt JPEG data: 2442 extraneous bytes before marker 0xd9
Corrupt JPEG data: 172 extraneous bytes before marker 0xbf
Corrupt JPEG data: 1380 extraneous bytes before marker 0x64
Corrupt JPEG data: 206 extraneous bytes before marker 0x89
Corrupt JPEG data: 496 extraneous bytes before marker 0xd9
Corrupt JPEG data: 189 extraneous bytes before marker 0x8a
Corrupt JPEG data: 3756 extraneous bytes before marker 0xd9
Corrupt JPEG data: 2799 extraneous bytes before marker 0xd9
Corrupt JPEG data: 370 extraneous bytes before marker 0x44
Corrupt JPEG data: 914 extraneous bytes before marker 0xd9
Corrupt JPEG data: 2188 extraneous bytes before marker 0xd9
Corrupt JPEG data: 254 extraneous bytes before marker 0xee
Corrupt JPEG data: 1546 extraneous bytes before marker 0xd9
Corrupt JPEG data: 4571 extraneous bytes before marker 0xd9
Corrupt JPEG data: 437 extraneous bytes before marker 0xd6
Corrupt JPEG data: 1295 extraneous bytes before marker 0xc0
Corrupt JPEG data: 1430 extraneous bytes before marker 0xd9
Corrupt JPEG data: 4626 extraneous bytes before marker 0xd9
Corrupt JPEG data: 2537 extraneous bytes before marker 0xd9
Corrupt JPEG data: 802 extraneous bytes before marker 0x10
Corrupt JPEG data: 185 extraneous bytes before marker 0x71
Corrupt JPEG data: 158 extraneous bytes before marker 0xa4
Corrupt JPEG data: 4205 extraneous bytes before marker 0xd9
Corrupt JPEG data: 90 extraneous bytes before marker 0xbb
Corrupt JPEG data: 41 extraneous bytes before marker 0x66
Corrupt JPEG data: 417 extraneous bytes before marker 0xd9
Corrupt JPEG data: 489 extraneous bytes before marker 0x04
Corrupt JPEG data: 102 extraneous bytes before marker 0x3a
Corrupt JPEG data: 219 extraneous bytes before marker 0xc6
Corrupt JPEG data: 4207 extraneous bytes before marker 0xd9
Corrupt JPEG data: 8 extraneous bytes before marker 0x0a
Corrupt JPEG data: 226 extraneous bytes before marker 0xfc
Corrupt JPEG data: 2264 extraneous bytes before marker 0xd9
Corrupt JPEG data: 808 extraneous bytes before marker 0xc0
Corrupt JPEG data: 533 extraneous bytes before marker 0xce
Corrupt JPEG data: 4952 extraneous bytes before marker 0xd9
Corrupt JPEG data: 364 extraneous bytes before marker 0xd8
Corrupt JPEG data: 176 extraneous bytes before marker 0x98
Corrupt JPEG data: 3790 extraneous bytes before marker 0xd9
Corrupt JPEG data: 186 extraneous bytes before marker 0x58
Corrupt JPEG data: 326 extraneous bytes before marker 0xb5
Corrupt JPEG data: 541 extraneous bytes before marker 0x2c
Corrupt JPEG data: 61 extraneous bytes before marker 0x59
Corrupt JPEG data: 380 extraneous bytes before marker 0x11
Corrupt JPEG data: 270 extraneous bytes before marker 0xe2
Corrupt JPEG data: 3350 extraneous bytes before marker 0xd9
Corrupt JPEG data: 39 extraneous bytes before marker 0x2a
Corrupt JPEG data: 117 extraneous bytes before marker 0x9c
Corrupt JPEG data: 3960 extraneous bytes before marker 0xd9
Corrupt JPEG data: 2509 extraneous bytes before marker 0xd9
Corrupt JPEG data: 4742 extraneous bytes before marker 0xd9
Corrupt JPEG data: 1171 extraneous bytes before marker 0xf0
Corrupt JPEG data: 479 extraneous bytes before marker 0x8c
Corrupt JPEG data: 54 extraneous bytes before marker 0xd6
Corrupt JPEG data: 3583 extraneous bytes before marker 0xd9
Corrupt JPEG data: 1852 extraneous bytes before marker 0xa6
Corrupt JPEG data: 98 extraneous bytes before marker 0x94
Corrupt JPEG data: 84 extraneous bytes before marker 0xc5
Corrupt JPEG data: 4021 extraneous bytes before marker 0xd9
Corrupt JPEG data: 211 extraneous bytes before marker 0x2d
Corrupt JPEG data: 25 extraneous bytes before marker 0x2d
Corrupt JPEG data: 30 extraneous bytes before marker 0x55
Corrupt JPEG data: 181 extraneous bytes before marker 0xa7
Corrupt JPEG data: 946 extraneous bytes before marker 0xd9
Corrupt JPEG data: 479 extraneous bytes before marker 0x17
Corrupt JPEG data: 103 extraneous bytes before marker 0x8c
Corrupt JPEG data: 4234 extraneous bytes before marker 0xd9
Corrupt JPEG data: 205 extraneous bytes before marker 0xc0
Corrupt JPEG data: 601 extraneous bytes before marker 0xd9
Corrupt JPEG data: 1199 extraneous bytes before marker 0x7c
Corrupt JPEG data: 142 extraneous bytes before marker 0x59
Corrupt JPEG data: 1904 extraneous bytes before marker 0xc4
Corrupt JPEG data: 498 extraneous bytes before marker 0xb9
Corrupt JPEG data: 31 extraneous bytes before marker 0xd8
Corrupt JPEG data: 522 extraneous bytes before marker 0xa1
Corrupt JPEG data: 2 extraneous bytes before marker 0x91
Corrupt JPEG data: 803 extraneous bytes before marker 0x98
Corrupt JPEG data: 184 extraneous bytes before marker 0xcf
Corrupt JPEG data: 4621 extraneous bytes before marker 0xd9
Corrupt JPEG data: 4271 extraneous bytes before marker 0xd9
Corrupt JPEG data: 30 extraneous bytes before marker 0x64
Corrupt JPEG data: 629 extraneous bytes before marker 0xa9
Corrupt JPEG data: 1323 extraneous bytes before marker 0xdd
Corrupt JPEG data: 2947 extraneous bytes before marker 0xd9
Corrupt JPEG data: 478 extraneous bytes before marker 0x7c
Corrupt JPEG data: 5067 extraneous bytes before marker 0xd9
Corrupt JPEG data: 313 extraneous bytes before marker 0xe0
Corrupt JPEG data: 4200 extraneous bytes before marker 0xd9
Corrupt JPEG data: 431 extraneous bytes before marker 0xd6
Corrupt JPEG data: 420 extraneous bytes before marker 0xbc
Corrupt JPEG data: 233 extraneous bytes before marker 0xf6
Corrupt JPEG data: 620 extraneous bytes before marker 0x71
Corrupt JPEG data: 189 extraneous bytes before marker 0xe6
Corrupt JPEG data: 656 extraneous bytes before marker 0x61
Corrupt JPEG data: 35 extraneous bytes before marker 0x92
Corrupt JPEG data: 5107 extraneous bytes before marker 0xd9
Corrupt JPEG data: 808 extraneous bytes before marker 0xd5
Corrupt JPEG data: 2715 extraneous bytes before marker 0xd9
Corrupt JPEG data: 14 extraneous bytes before marker 0xb3
Corrupt JPEG data: 109 extraneous bytes before marker 0xa1
Corrupt JPEG data: 3360 extraneous bytes before marker 0xd9
Corrupt JPEG data: 816 extraneous bytes before marker 0x65
Corrupt JPEG data: 934 extraneous bytes before marker 0x79
Corrupt JPEG data: 390 extraneous bytes before marker 0x23
Corrupt JPEG data: 3264 extraneous bytes before marker 0xd9
Corrupt JPEG data: 77 extraneous bytes before marker 0xd9
Corrupt JPEG data: 776 extraneous bytes before marker 0xcb
Corrupt JPEG data: 43 extraneous bytes before marker 0x8c
```

Don't forget the commit hash in versions please.

**Comment by mergen3107** [comment]
It’s the nightly and stable from OTA

**Comment by mergen3107** [comment]
So nightly must be v2024.07-25-g5affb6823_2024-07-21/

**Comment by poire-z** [comment]
I also don't get images with current master. They are there with v2024.01 (only appimage at hands).
Will be hard to bissect with the recent big build changes...
Images are stored as base64 inside the fb2, so I would suspect either a XML parser changes or a LVStream one.

**Comment by benoit-pierre** [comment]
The culprit is 9d70a2e588ad063b550bc279ad3cffd9ff8e40cc.

**Comment by benoit-pierre** [comment]
Ping @bbshelper.

**Comment by benoit-pierre** [comment]
Anyway, can't revert just that commit (2 other more conflict with it), but reverting `lvxml.cpp` contents to 07cbc66748ff17b63723bdff50c3489c004f4821 work.

**Comment by benoit-pierre** [comment]
@mergen3107: see https://github.com/koreader/crengine/pull/583 for a quick fix.

**Comment by mergen3107** [comment]
Thank you!

**Comment by mergen3107** [comment]
Interestingly enough, some fb2 books show images just fine. I’ll upload an example later

**Comment by mergen3107** [comment]
Tested on emulator so far, the revert works! Thanks again @benoit-pierre 
I also noticed that fb2 covers in FM are also missing in the current build (without revert)

**Comment by bbshelper** [comment]
I'll take a look.

**Comment by bbshelper** [comment]
@mergen3107 would you try https://github.com/koreader/crengine/pull/584 ? I can confirm the regression but somehow I can't reproduce the issue with my local emulator.

**Comment by mergen3107** [comment]
@bbshelper 
> somehow I can't reproduce the issue with my local emulator.

I had success by doing `./kodev clean` between builds :D 

Thank you! It worked in emulator!
I put it into Kindle, and it didn't work at first.
Then I tried to rename the cache file, and voilà! It worked after it rebuild the cache.
The worng cache weighs around 1.7 MB, new one weighs around 20 MB (which is expected, there are TONS of images in this book).

I attach it here for science (remove `.txt`) :P 

[Boris_Akunin-lesnitsa.fb2.066bc0af.1.cr3.txt](https://github.com/user-attachments/files/16378699/Boris_Akunin-lesnitsa.fb2.066bc0af.1.cr3.txt)

**Comment by mergen3107** [comment]
Closed by #12360
